source/pervasive/std_specs/vec.rs

```diff
@@ -220,5 +222,31 @@ pub fn ex_vec_as_slice<T, A: Allocator>(vec: &Vec<T, A>) -> (slice: &[T])
     vec.as_slice()
 }
 
+#[verifier::external_fn_specification]
+pub fn ex_vec_split_off<T, A: Allocator+ std::clone::Clone>(vec: &mut Vec<T, A>, at: usize) -> (return_value: Vec<T, A>)
+    ensures
+        vec@ == old(vec)@.subrange(0,at as int),
+        return_value@ == old(vec)@.subrange(at as int, old(vec)@.len() as int),
+{
+    vec.split_off(at)
+}
+
+#[verifier::external_fn_specification]
+pub fn ex_vec_truncate<T, A: Allocator>(vec: &mut Vec<T, A>, len: usize)
+    ensures
+        len <= vec.len() ==> vec@ == old(vec)@.subrange(0,len as int),
+        len > vec.len() ==> vec@ == old(vec)@,
+{
+    vec.truncate(len)
+}
+
+// #[verifier::external_fn_specification]
+// pub fn ex_vec_to_vec<T: std::clone::Clone, A: Allocator>(vec: &mut Vec<T, A>) -> (return_value: Vec<T, ExGlobal>)
+//     ensures
+//         vec@ == old(vec)@,
+//         return_value@ =~= old(vec)@,
+// {
+//     vec.to_vec()
+// }
```

Comment:
Why is this commented out?

Reply:
Oh I forgot about this cause I did it very early on, I think I got stuck and confused on it. Anyways I'm just going to delete it because it's not really relevant to this project anyways